src/main/java/org/opensearch/knn/indices/ModelMetadata.java

```diff
@@ -89,9 +89,13 @@ public ModelMetadata(
         this.knnEngine = Objects.requireNonNull(knnEngine, "knnEngine must not be null");
         this.spaceType = Objects.requireNonNull(spaceType, "spaceType must not be null");
         int maxDimensions = KNNEngine.getMaxDimensionByEngine(this.knnEngine);
-        if (dimension <= 0 || dimension >= maxDimensions) {
```

Comment:
This can cause regression for some one already have indices with dimensions=maxDimensions. Can we revert to original?

Reply:
It will currently fail if maxDimensions == dimension. So I switched to dimension > maxDimension

Reply:
I don't think it will fail, current max is 10.000, new is 16.000. So with cx dimensions = 10.000 it will still pass new check.